scripts/hid_configurator/gui/gui.py

```diff
@@ -0,0 +1,145 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+
+import threading
+from kivy.app import App
+from kivy.clock import mainthread
+from kivy.config import Config
+from kivy.properties import ObjectProperty
+from kivy.uix.button import Button
+from kivy.uix.floatlayout import FloatLayout
+from kivy.uix.gridlayout import GridLayout
+from kivy.uix.popup import Popup
+
+from gui_backend import Device
+
+
+class MouseOptions(GridLayout):
+    pass
+
+
+class DfuButtons(GridLayout):
+    def start_dfu_thread(self):
+        app_ref = App.get_running_app()
+        threading.Thread(target=app_ref.dfu, args=(self.update_progressbar,)).start()
+
+    @mainthread
+    def update_progressbar(self, permil):
+        self.ids.pb.value = permil / 10 + 0.1
+
+
+class DropDownButton(Button):
+    pass
+
+
+class Gui(App):
+    def add_mouse_settings(self):
+        mouse_options = MouseOptions()
+        self.root.ids.possible_settings_place.add_widget(mouse_options)
+
+    def add_dfu_buttons(self):
+        dfu_buttons = DfuButtons()
+        self.root.ids.dfu_buttons_place.add_widget(dfu_buttons)
+
+    def clear_possible_settings(self):
+        print("Clear possible settings")
+        self.root.ids.possible_settings_place.clear_widgets()
+        self.root.ids.dfu_buttons_place.clear_widgets()
+
+    def setcpi_callback(self, value):
+        self.device.setcpi(value)
+
+    def fetchcpi_callback(self):
+        returned = self.device.fetchcpi()
+        if returned:
+            return returned
+        else:
+            return 0
+
+    def initialize_dropdown(self):
+        device_list = Device.list_devices()
+        for device in device_list:
+            self.add_dropdown_button(device)
+
+    def initialize_device(self, device_type):
+        self.device = Device(device_type)
+        print("Initialize {}".format(device_type))
+
+    def show_possible_settings(self):
```

Comment:
Later we should also add desktop mice here

Reply:
Yes, I agree